nx/lib/nx/defn/expr.ex

```diff
@@ -152,18 +152,23 @@ defmodule Nx.Defn.Expr do
 
     clauses = Enum.zip(preds, exprs)
 
-    out = flatten_to_composite(out, context, exprs, &expr(&1, context, :cond, [clauses, last]))
+    out =
```

Comment:
I had to keep this part here because the output vectorized axes change in relation to the input, but the actual vectorization happens inside flatten_to_composite now